# HG changeset patch
# User Rik <rik@octave.org>
# Date 1533839291 25200
#      Thu Aug 09 11:28:11 2018 -0700
# Node ID ecdced9fe445cd40e4be434f8a89a9eb54707e3e
# Parent  0f6c81ca512899ed04302c69e96c20f7aab91a60
flipdim.m: Make m-file a legacy function.

* NEWS: Add flipdim to list of legacy functions.

* scripts/general/module.mk, scripts/legacy/module.mk: Move flipdim.m
from general/ directory to legacy/ directory.

* func.tst: Change instances of flipdim to flip

* flipdim.m: Add Octave:legacy-function warning.  Remove all BIST tests.
Update docstring.

diff --git a/NEWS b/NEWS
--- a/NEWS
+++ b/NEWS
@@ -71,16 +71,17 @@ Summary of important user-visible change
 
     The following functions have been declared legacy functions which
     means they are obsolete and should not be used in any new code.
     Unlike deprecated functions, however, their removal from Octave has
     not yet been scheduled.
 
       Function             | Replacement
       ---------------------|------------------
+      flipdim              | flip
       isequalwithequalnans | isequaln
       isstr                | ischar
       setstr               | char
       strmatch             | strncmp or strcmp
 
  ** Deprecated functions.
 
     The following functions have been deprecated in Octave 5.0 and will
diff --git a/scripts/general/module.mk b/scripts/general/module.mk
--- a/scripts/general/module.mk
+++ b/scripts/general/module.mk
@@ -23,17 +23,16 @@ FCN_FILE_DIRS += \
   %reldir%/cumtrapz.m \
   %reldir%/curl.m \
   %reldir%/dblquad.m \
   %reldir%/deal.m \
   %reldir%/deg2rad.m \
   %reldir%/del2.m \
   %reldir%/divergence.m \
   %reldir%/flip.m \
-  %reldir%/flipdim.m \
   %reldir%/fliplr.m \
   %reldir%/flipud.m \
   %reldir%/gradient.m \
   %reldir%/idivide.m \
   %reldir%/int2str.m \
   %reldir%/integral.m \
   %reldir%/integral2.m \
   %reldir%/integral3.m \
diff --git a/scripts/general/flipdim.m b/scripts/legacy/flipdim.m
rename from scripts/general/flipdim.m
rename to scripts/legacy/flipdim.m
--- a/scripts/general/flipdim.m
+++ b/scripts/legacy/flipdim.m
@@ -15,25 +15,23 @@
 ##
 ## You should have received a copy of the GNU General Public License
 ## along with Octave; see the file COPYING.  If not, see
 ## <https://www.gnu.org/licenses/>.
 
 ## -*- texinfo -*-
 ## @deftypefn  {} {} flipdim (@var{x})
 ## @deftypefnx {} {} flipdim (@var{x}, @var{dim})
-## Flip array across dimension @var{dim}.
-##
-## This function is an alias for @code{flip} and exists for backwards and
-## @sc{matlab} compatibility.  See @code{flip} for complete usage information.
-##
+## This function is obsolete.  Use @code{flip} instead.
 ## @seealso{flip, fliplr, flipud, rot90, rotdim}
 ## @end deftypefn
 
-## Author: David Bateman, Jaroslav Hajek
+function y = flipdim (varargin)
 
-function y = flipdim (varargin)
+  persistent warned = false;
+  if (! warned)
+    warned = true;
+    warning ("Octave:legacy-function",
+             "flipdim is obsolete; please use flip instead");
+  endif
+
   y = flip (varargin{:});
 endfunction
-
-
-## No tests needed for alias.  All tests for functionality are in flip.m
-%!assert (1)
diff --git a/scripts/legacy/module.mk b/scripts/legacy/module.mk
--- a/scripts/legacy/module.mk
+++ b/scripts/legacy/module.mk
@@ -1,11 +1,12 @@
 FCN_FILE_DIRS += scripts/legacy
 
 %canon_reldir%_FCN_FILES = \
+  %reldir%/flipdim.m \
   %reldir%/isequalwithequalnans.m \
   %reldir%/isstr.m \
   %reldir%/setstr.m \
   %reldir%/strmatch.m
 
 %canon_reldir%dir = $(fcnfiledir)/legacy
 
 %canon_reldir%_DATA = $(%canon_reldir%_FCN_FILES)
diff --git a/test/func.tst b/test/func.tst
--- a/test/func.tst
+++ b/test/func.tst
@@ -158,19 +158,19 @@
 %! __fntestfunc__ ("fliplr", m1);
 %!test
 %! __fntestfunc__ ("fliplr", m3);
 %!test
 %! __fntestfunc__ ("flipud", m1);
 %!test
 %! __fntestfunc__ ("flipud", m3);
 %!test
-%! __fntestfunc__ ("flipdim", m1, 2);
+%! __fntestfunc__ ("flip", m1, 2);
 %!test
-%! __fntestfunc__ ("flipdim", m3, 2);
+%! __fntestfunc__ ("flip", m3, 2);
 %!test
 %! __fntestfunc__ ("transpose", m1);
 %!test
 %! __fntestfunc__ ("transpose", m3);
 %!test
 %! __fntestfunc__ ("ctranspose", m1);
 %!test
 %! __fntestfunc__ ("ctranspose", m3);
